onnx/shape_inference/implementation.cc

```diff
@@ -185,6 +185,30 @@ static void InferShapesImpl(
   std::unordered_map<std::string, const TensorProto*> inputDataByName;
   for (const auto& tp : g->initializer()) {
     inputDataByName[tp.name()] = &tp;
+    
+    // This is for testing new IR: some tensors can only exist in initializer and not in input
+    // So shape_inference should make use of initializer shapes
+    // Store initializer shape info in value_info as well
+    TypeProto_Tensor* newTensorFromInitializer;
+    auto iter = valueTypesByName.find(tp.name());
+    // If it already exists in input, simply use the original input
+    if (iter != valueTypesByName.end()) {
+      newTensorFromInitializer = valueTypesByName[tp.name()]->mutable_tensor_type();
```

Comment:
If such tenor has been found in input, maybe we don't need to do anything here from the initializer?